src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1023,17 +1023,32 @@ private void DeleteIndexInternal(Index index, bool raiseNotification = true)
                 });
             }
 
-            // we always want to try to delete the directories
-            // because Voron and Periodic Backup are creating temp ones
-            //if (index.Configuration.RunInMemory)
-            //    return;
-
             var name = IndexDefinitionBase.GetIndexNameSafeForFileSystem(index.Name);
 
             var indexPath = index.Configuration.StoragePath.Combine(name);
 
             var indexTempPath = index.Configuration.TempPath?.Combine(name);
 
+            if (index.Configuration.RunInMemory)
+            {
+                // when running in-memory all storage environment files are temporary so deleted on close but
+                // we want to delete the empty directories as well
+                
+                // but when running a replacement index we cannot stop environment and move its directory so we leave it as is
+                // we can have temp files of different environments in the same temp dir (when index is updated multiple times)
+                // so let's delete the directory only if there are no other files inside - last dispose will delete the dir
+
```

Comment:
Don't we have to worry about time of check vs. time of use?
So we don't have any files there, but then it is created?

Reply:
I don't think so. This is very specific issue to replacement indexes. Given that the folder is empty it means that we were the last "replacement" which has been just disposed and a new one wasn't created (if it was the it would be put in the same dir so we'd still have files here)